src/features/user/UserProfile/components/RegisterTrees.tsx

```diff
@@ -0,0 +1,327 @@
+import React from 'react';
+import styles from '../styles/RegisterModal.module.scss';
+import i18next from '../../../../../i18n';
+import MapGL, {
+  FlyToInterpolator,
+  Marker,
+  NavigationControl,
+  WebMercatorViewport,
+} from 'react-map-gl';
+import * as d3 from 'd3-ease';
+import BackButton from '../../../../../public/assets/images/icons/BackButton';
+import { useRouter } from 'next/router';
+import { getAuthenticatedRequest, postAuthenticatedRequest, putAuthenticatedRequest } from '../../../../utils/apiRequests/api';
+import UploadImagesStep from './RegisterTrees/UploadImages';
+import { Step, StepContent, StepLabel, Stepper } from '@material-ui/core';
+import dynamic from 'next/dynamic';
+import MaterialTextField from '../../../common/InputTypes/MaterialTextField';
+import { DatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers';
+import { Controller, useForm } from 'react-hook-form';
+import DateFnsUtils from '@date-io/date-fns';
+import { localeMapForDate } from '../../../../utils/language/getLanguageName';
+import getStoredConfig from '../../../../utils/getConfig/getStoredConfig';
+import SingleContribution from './RegisterTrees/SingleContribution';
+
+const DrawMap = dynamic(() => import('./RegisterTrees/DrawMap'), {
+  ssr: false,
+  loading: () => <p></p>,
+});
+
+interface Props {
+  slug: any;
+  session: any;
+}
+
+const { useTranslation } = i18next;
+export default function RegisterTrees({ slug, session
+}: Props) {
+  const router = useRouter();
+  const { t } = useTranslation(['me', 'common']);
+  const [isMultiple, setIsMultiple] = React.useState(false);
+  const [contributionGUID, setContributionGUID] = React.useState('');
+  const [contributionDetails, setContributionDetails] = React.useState({});
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const screenWidth = window.innerWidth;
+  const isMobile = screenWidth <= 767;
+  const defaultMapCenter = isMobile ? [22.54, 9.59] : [36.96, -28.5];
+  const defaultZoom = isMobile ? 1 : 1.4;
+  const [plantLocation, setplantLocation] = React.useState();
+  const [geometry, setGeometry] = React.useState();
+  const [viewport, setViewPort] = React.useState({
+    height: '100%',
+    width: '100%',
+    latitude: defaultMapCenter[0],
+    longitude: defaultMapCenter[1],
+    zoom: defaultZoom,
+  });
+  const [userLang, setUserLang] = React.useState('en');
+  const [countryBbox, setCountryBbox] = React.useState();
+  const [registered, setRegistered] = React.useState(false);
+
+  React.useEffect(() => {
+    if (localStorage.getItem('language')) {
+      let userLang = localStorage.getItem('language');
+      if (userLang) setUserLang(userLang);
+    }
+
+    async function getUserCountryBbox() {
+      var country = getStoredConfig('country');
+      const result = await fetch(`http://api.mapbox.com/geocoding/v5/mapbox.places/${country}.json?types=country&limit=1&access_token=${process.env.MAPBOXGL_ACCESS_TOKEN}`);
```

Comment:
Not using HTTPS here fails when using the website with SSL: `RegisterTrees.tsx?ed5c:69 Mixed Content: The page at 'https://localhost/register-trees' was loaded over HTTPS, but requested an insecure resource 'http://api.mapbox.com/geocoding/v5/mapbox.places/DE.json?types=country&limit=1&access_token=...'. This request has been blocked; the content must be served over HTTPS.`

Reply:
We never before used a direct API call for Mapbox - isn't this functionality available by any npm module we already use?

Reply:
do we have to support users using HTTP?
no, the npm modules only have map libraries. there's a different library for geocoding API `@mapbox/mapbox-gl-geocoder`

Reply:
HTTP would only be useful for development running with `npm run dev` - I am using HTTPS locally by using a local proxy server called caddy to work with SSL.